File: PROGRAMMERS/clothes.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

map<string, int> m;
int solution(vector<vector<string>> clothes) {
    int answer = clothes.size();
    for(int i=0 ; i<clothes.size() ; i++){
        m[clothes[i][1]]++;
    }
    auto it = m.begin();
    answer = it->second;
    for(it++ ; it != m.end() ; it++){
        answer+=(answer+1)*it->second;
    }
    return answer;
}

File: PROGRAMMERS/Clothes.java
import java.util.*;
class Clothes {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> hm = new HashMap<>();
        
        for(int i=0 ; i<clothes.length ; i++){
            hm.put(clothes[i][1], hm.getOrDefault(clothes[i][1], 0) + 1);
        }
        
        Set<String> ks = hm.keySet();
        for(String key : ks){
            answer *= hm.get(key) + 1;
        }
        return answer-1;
    }
}
